crates/algorithms/src/haversine/mod.rs

```diff
@@ -0,0 +1,131 @@
+use core::fmt::Debug;
+use nalgebra::Point2;
+use num_traits::Float;
+
+///calculate distance between point with use haversine formula
+/// return distance in km
+#[cfg_attr(
+    feature = "tracing",
+    tracing::instrument("Calculate Haversine Data", skip_all)
+)]
+fn calculate_haversine_distance<T>(p1: Point2<T>, p2: Point2<T>, sphere_radius: T) -> T
+where
+    T: Copy + Float + Debug + 'static,
+{
+    //define two = 2 to support generic
+    let two = T::from(2.0).unwrap();
```

Comment:
We should really carefully avoid using unwrap()s, even if it is in a controlled scenario^
How about something like let two = T::one() + T::one();?